src/core/endpoint.ts

```diff
@@ -0,0 +1,32 @@
+import type { ILogObj, Logger } from 'tslog';
+import type Service from '../modules/apiService';
+import type { IApiError } from '../modules/IService';
+
+export default class Endpoint {
+  protected service: Service;
+
+  protected log: Logger<ILogObj>;
+
+  constructor(serviceApi: Service, logger: Logger<ILogObj>) {
+    this.service = serviceApi;
+    this.log = logger;
+  }
+
+  static isApiError(response: IApiError | any): response is IApiError {
+    return response.errorCode !== undefined;
+  }
+
+  static makeQueryStringObject = (obj: any) => {
+    const newObjArray: string[][] = [];
+
+    Object.entries(obj).forEach(([key, value]: [string, any]) => {
```

Comment:
what about this any

Reply:
This could literally be anything. You don't know what is going to be passed in as parameters from calling code. That is why the types are explicitly checked later (at runtime vs at compile).